spec/models/shelter_spec.rb

```diff
@@ -41,31 +56,56 @@
         expect(Shelter.order_by_number_of_pets).to eq([@shelter_1, @shelter_3, @shelter_2])
       end
     end
+
+    describe "#order_by_reverse_alphabetically" do
+      it "returns all Shelters listed in reverse alphabetical order by name" do
+        @turing = Shelter.create(foster_program: true, name: "Turing", city: "Backend", rank: 3)
+        @fsa = Shelter.create(foster_program: true, name: "Fullstack Academy", city: "Backend", rank: 3)
+        @codesmith = Shelter.create(foster_program: true, name: "Codesmith", city: "Backend", rank: 3)
+        @rithm = Shelter.create(foster_program: true, name: "Rithm School", city: "Backend", rank: 3)
+        @hackreactor = Shelter.create(foster_program: true, name: "Hack Reactor", city: "Backend", rank: 3)
+
+        @shelters = Shelter.order_by_reverse_alphabetically
+        first = @shelters.first
+        last = @shelters.last
+
+        expect(@shelters).to eq [@turing, @rithm, @shelter_2, @hackreactor, @fsa, @shelter_3, @codesmith, @shelter_1]
+        expect(first.name).to eq("Turing")
+        expect(last.name).to eq("Aurora shelter")
+      end
+    end
+
+    describe "#pending_applications" do
+      it "returns a list of shelters that have pending applications" do
+        expect(Shelter.pending_applications).to eq ([@shelter_1, @shelter_3])
+      end
+    end
   end
 
   describe "instance methods" do
-    describe ".adoptable_pets" do
+    describe "#pet_count" do
+      it "returns the number of pets at the given shelter" do
+        expect(@shelter_1.pet_count).to eq(3)
+      end
+    end
+
+    describe "#adoptable_pets" do
       it "only returns pets that are adoptable" do
         expect(@shelter_1.adoptable_pets).to eq([@pet_2, @pet_4])
       end
     end
 
-    describe ".alphabetical_pets" do
+    describe "#alphabetical_pets" do
       it "returns pets associated with the given shelter in alphabetical name order" do
         expect(@shelter_1.alphabetical_pets).to eq([@pet_4, @pet_2])
       end
     end
 
-    describe ".shelter_pets_filtered_by_age" do
+    describe "#shelter_pets_filtered_by_age" do
       it "filters the shelter pets based on given params" do
         expect(@shelter_1.shelter_pets_filtered_by_age(5)).to eq([@pet_4])
       end
     end
-
-    describe ".pet_count" do
-      it "returns the number of pets at the given shelter" do
-        expect(@shelter_1.pet_count).to eq(3)
-      end
-    end
   end
+
```

Comment:
Your testing overall is excellent! At both the unit (model spec) and integration (feature spec) level, you're capturing reasonable sad paths and watching for logical edge cases. Nice work!